src/resources/views/base/inc/alerts.blade.php

```diff
@@ -1,21 +1,40 @@
 {{-- Bootstrap Notifications using Prologue Alerts & PNotify JS --}}
 <script type="text/javascript">
-  Noty.overrideDefaults({
-    layout   : 'topRight',
-    theme    : 'backstrap',
-    timeout  : 2500, 
-    closeWith: ['click', 'button'],
-  });
+    // This is intentionaly run after dom loads so this way we can avoid showing duplicate alerts
+    // when the user is beeing redirected by persistent table, that happens before this event triggers.
+    window.addEventListener('DOMContentLoaded', function() {
+        Noty.overrideDefaults({
+            layout: 'topRight',
+            theme: 'backstrap',
+            timeout: 2500,
+            closeWith: ['click', 'button'],
+        });
 
-  @foreach (\Alert::getMessages() as $type => $messages)
+        // get alerts from the alert bag
+        var $alerts_from_php = JSON.parse('@json(\Alert::getMessages())');
```

Comment:
@pxpm I'm only able to make this work by adding 
```suggestion
        var $alerts_from_php = JSON.parse('@json(\Alert::getMessages())');
        @php
        \Alert::flush();
        @endphp
```
Otherwise Alerts are shown twice, can you please check?

Reply:
@promatik I could not replicate when this is needed. 

I'v just removed and I think I went through all cases I know and I didn't get repeated alerts. 

How can I reproduce the repeated alerts ? 

This is what I'v tried:

- setup a custom save action:
```php
$this->crud->replaceSaveActions([
            'name' => 'save_action_one',
            'redirect' => function($crud, $request, $itemId) {
               return $crud->route;
            },
            'button_text' => 'Custom save message', 
            'visible' => function($crud) {
                return true;
            }, 
        ]);
```
- Filtered some crud, and then go to create. Created 1 entry and the save action redirected me to the index (list), as I had previously filtered crud and persistence is enable I am going to be redirected to my filtered table via persistent table. 

Result: Show that the entry was created and save action changed. 2 notifications (didn't show before redirecting)

Used default save actions, tried save and back, save and edit, save and new, every time 2 notifications, entry created, default action changed when I did it.

If in backpack demo you change to production, you will get some extra alerts when creating a monster for example.

Those don't show duplicated:

![image](https://user-images.githubusercontent.com/7188159/107267794-451dd900-6a3f-11eb-9dda-a6a5dbfb5f38.png)

So yeah :D help me here to reproduce, I have no doubts I could have missed something here, so please @promatik double-check because you were probably right, but I would like to test it and I can't 🤦 

Best,
Pedro

Reply:
@pxpm it was simple, I added the Alerts on the CrudController Setup, so it is fired on every Operation (List, Reorder, Form ...)
Then if you are in any Operation, other than List, the double Alert happens, let me know if you were able to reproduce it ...

Reply:
@promatik I found the oposite 👆 

I added to `setup()` 

```php
 \Alert::add('info', 'omg what is happening')->flash();
 \Alert::add('success', 'omg what is happening')->flash();
```

If I go to create or go to List without persistent table redirect I get this (expected)

![image](https://user-images.githubusercontent.com/7188159/107404135-deabc000-6afd-11eb-9777-6e8bff999ff1.png)

If I go to List, with persistent table redirect I get this: (expect the unexpected)

![image](https://user-images.githubusercontent.com/7188159/107404375-216d9800-6afe-11eb-8af2-ef8fd06db5f3.png)

Events:
- We setup list, doing this we run `setup()` and add 2 alerts. 
- List renders until datatables found we are going to be redirected by persistence, so alerts saved to LS and redirect
- We setup list, doing this we run `setup()` and add 2 alerts.
- We get previous 2 alerts from LS, and merge with our 2 alerts from the new rebuild.

I think this only happens because we are using alerts in an "unusual" manner, atleast everytime I see alerts beeing added it's in response to some action to be used in the next request. Here we are testing setting alerts in EVERY request. So, yeah, when redirected by persistent-table is a new request, the ALL requests alerts are going to be set again.

Is it expected to be used like this ? If so, we might need to do some `text` comparison to not add the second alert, because using your suggestion seems to not fix the issue. 

Can you test it out? 

Best,
Pedro

Reply:
@pxpm understood 👌
Probably the last time I was just testing the "refresh", because on refresh the problem doesn't happen, don't know 🤷‍♂️
Anyway, you're right, that solution doesn't solve it...

Text comparison seems like an overkill operation, but `Alerts` like these may be used, for instance our demo, fires an alerts in during 3 minutes before and after refresh. We use javascript to do it, one may use a Controller to fire events like these 🤷‍♂️

So I think we should solve it. I'll leave some code suggestions...
Let me know what you think.